source/Halibut/Transport/ConnectionManager.cs

```diff
@@ -119,6 +127,18 @@ void OnConnectionDisposed(IConnection connection)
             }
         }
 
+        void SafelyDisposeConnection(IConnection connection, ILog log)
+        {
+            try
+            {
+                connection?.Dispose();
+            }
+            catch (Exception ex)
+            {
+                log?.WriteException(EventType.Error, "Exception disposing connection from pool", ex);
```

Comment:
Any thoughts what might cause this? I'm thinking along the lines of trying to have actionable errors logged if possible
Or is this a "just in case"?

Reply:
This is just in case, I've never seen one in the wild

Reply:
We don't do this anywhere else, and if `log` is null there will be other problems.